[Elizazaa] ip

elizazaa opened the pull request:
# NickiMinaj 🎤

> *"Queens don't compete with losers"* – Nicki Minaj

NickiMinaj is your personal chatbot assistant that keeps you on track. It's:

- **Conversational and intuitive**
- **Easy to interact with**
- **Super fast and reliable**

All you need to do is:

1. Download it from [here](https://github.com/your-repo/nickiminaj).
2. Run the application.
3. Start chatting with **NickiMinaj**.
4. Let it assist you in your daily tasks! 🤖

And it is **FREE**! 🎉

### Features:
✅ Real-time chat  
⬜ Advanced task management (coming soon)  
⬜ Smart reminders (coming soon)  

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        new NickiMinaj("./data/nickiminaj.NickiMinaj.txt").run();
    }
}
```


### **How to Use:**
1. **Copy** everything from the box above.
2. **Paste** it into your **PR description**.
3. Modify the content if needed.
4. Click **Save** to update the PR.

This keeps the formatting similar to your example but customized for **NickiMinaj**! Let me know if you'd like any changes. 🚀🎤

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
changed what duke class prints out in the main method
renamed the class and the file NickiM

Commit:
Changed class name

Commit:
Changed Name

Commit:
Revert "Changed Name"

This reverts commit 3b285c3d99a38083884decb8149a80e24425963b.

Commit:
Echoes inputs using scanner import

Commit:
imported ArrayList and List to use for making the toDo list

Commit:
fixed bugs in converting List toDo into the string output as specified in level 2

Commit:
changed output for the chatbot Responses

Commit:
Created Task.java

Commit:
added functions and constructor into task class

Commit:
integrated Task into NickiMinaj
- changed List type
- changed outputs

Commit:
created 3 subclasses

Commit:
added constructors and fields in deadline, event and todo
updated the main method in NickiMinaj to include the keywords todo, deadline, event

Commit:
fixed class cast exception bug from NickiMinaj main method

Commit:
fixed list formatting

Commit:
Updated input and expected for the the new inputs and expected outputs

Commit:
changed expected outputs accordingly (newlines/spaces were fixed)

Commit:
created Duke.java and modified NickiMinaj to handle errors using try catch

Commit:
handled more errors + added delete in the if else block

Commit:
create NickiMinaj.txt on ./data path

Commit:
Initialise storage and toDo as global variables

Commit:
Update storage.java

Commit:
add constructor in Task.java file to contain isDone boolean variable

Commit:
Revert "add constructor in Task.java file to contain isDone boolean variable"

This reverts commit d2547ac25d4a7103b489ec770a4c455b62f3aee1.

Commit:
Add constructor to Task.java file with isDone boolean variable

Commit:
Make Task class abstract with a new abstract method serialize();

Commit:
feat: Add serialize method to all subclasses of Task

Commit:
feat: add constructor to all subclasses of Task

Commit:
feat: Remove second assignment of toDo

Commit:
refactor: Change description into des (as per Java standard)

Commit:
feat: add saveTask method in Storage

This method is used so that the tasks that was entered as input (aka by
the user) will be written into NickiMinaj.txt

Commit:
Test: test NickiMinaj class with input

Commit:
feat: use localDateTime class to reformat Deadline and Event class

Commit:
feat: Add formatter in Deadline class using DateTimeFormatter

Commit:
fix: Fix compatability issues between NickiMinaj class and Deadline

The Deadline class's second constructor had been modified to take in
LocalDateTime by instead of a String.

Commit:
feat: Enable event class to use LocalDateTime using LocalTimeFormatter

Commit:
fix: Fix compatability issues between NickiMinaj class and Event

The constructor's parameter type was changed. Hence, we need to make
changes in the NickiMinaj class, where the constructor is used, in
order to prevent type mismatch

Commit:
refactor: guide users into the correct input format

Commit:
test: Test new features against the expected output

nhocmt227 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,74 @@
+                        LocalDateTime.parse(parts[3], DateTimeFormatter.ofPattern("d/M/yyyy HHmm")),
+                        isDone);
+            case "E":
+                return new Event(parts[2],
```
You may consider following the java cding standard for switch and case commands.

nhocmt227 commented on src/main/java/NickiMinaj.java:
```diff
@@ -0,0 +1,113 @@
+        }
+        scanner.close();
+    }
+}
```
I think you have have good naming and style.

nhocmt227 commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,18 @@
+public class Todo extends Task {
```
I think you have done a good job in following the naming convention.

nhocmt227 reviewed (commented):
Second PR review

Commit:
Create 3 new classes: Parser, TaskList, Ui

Commit:
Refactor NickiMinaj class into 3 other classes using more OOP

Commit:
Fix bugs in NickiMinaj

Commit:
Add abstract Command class and its specific subclasses

Commit:
Refactor NickiMinaj class to not handle the different input cases

Commit:
Refactor the classes into respective packages

Commit:
Refactor the classes into respective packages

Commit:
Run NickiMinaj class to test expected outputs

Commit:
Merge branch 'master' of https://github.com/elizazaa/ip

* 'master' of https://github.com/elizazaa/ip:
  Fix bugs in NickiMinaj

# Conflicts:
#	src/main/java/NickiMinaj.java

Casielim commented on src/main/java/nickiminaj/Parser.java:
```diff
@@ -0,0 +1,36 @@
+        String[] parts = input.split(" ", 2);
+        String commandType = parts[0];
+
+        switch (commandType) {
```
IIRC, there is no need to have indentation for case clauses according to Java Coding standard.

Casielim reviewed (commented):
LGTM!
All method and variable names are clear to me. Just a minor suggestion regarding coding standards.😊

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

* origin/add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Commit:
Add JUnit Test for TaskList class

Commit:
Delete old NickiMinaj Class (without abstraction)

Commit:
Add ParserTest class to test Parser class

Commit:
Package the app as an executable JAR file

Commit:
Change output of unknown commands

from "OOPS!!! I'm sorry, but I don't know what that means :-(" to
"Girl... I don't even know what that means... but it sounds mad iconic."

Commit:
Add JavaDoc to Command class and its subclasses

Commit:
Create new FindCommand class to find keywords from a list

Commit:
Reformat Long line into 2 lines

Commit:
Change the for-loop format

Commit:
Change format of switch statements

Commit:
Add // Fallthrough comment for switch cases without break statement

Commit:
Removed the use of * for import statements

Commit:
Add JavaDoc to Task and its subclasses

Commit:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Removed the use of * for import statements
  Add // Fallthrough comment for switch cases without break statement
  Change format of switch statements
  Reformat Long line into 2 lines

# Conflicts:
#	src/main/java/nickiminaj/tasks/Todo.java

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Change the for-loop format
  Create new FindCommand class to find keywords from a list

Commit:
Fix bug for nickiMinaj

Commit:
Create files for nickiminaj's GUI

Commit:
Change outputs to match nickiminaj's personality

Commit:
Create new GUI for nickiminaj using FXML

Commit:
Make NickiMinaj chatbot GUI and CLI friendly

The NickiMinaj chatbot previously lacked proper integration between
the GUI and CLI. This commit improves the code structure to ensure
seamless functionality in both environments.

Changes made:
- Refactored `NickiMinaj.java` to handle both CLI and GUI interactions.
- Modified `Parser.java` to correctly process inputs from both
interfaces.
- Refactored `Command.java` to return output as a string instead of
printing directly to be compatible with GUI.

This update improves the user experience across different interfaces,
making the chatbot more versatile and maintainable.

Commit:
feat: Enhance NickiMinaj’s personality with more UI responses

NickiMinaj previously had limited responses, making interactions feel
less engaging. This commit improves the chatbot’s personality by adding
a wider variety of responses to enhance user experience.

Changes made:
- Expanded responses in `Ui.java` to include more expressive and witty
replies.
- Improved `Parser.java` handling to map inputs to diverse responses.
- Refactored `NickiMinaj.java` to integrate new UI responses dynamically.
- Updated `Command.java` to allow commands to generate varied feedback.
- Tweaked `InspireCommand.java` to provide more iconic Nicki Minaj quotes.

These enhancements make the chatbot feel more dynamic, entertaining,
and true to Nicki Minaj’s personality, providing a better user experience.

Commit:
Generate executable JAR for NickiMinaj chatbot

This commit adds the necessary configurations to build an executable
JAR file for the NickiMinaj chatbot.

Changes made:
- Configured `build.gradle` to include a `Main-Class` attribute.
- Applied the Shadow plugin to create a fat JAR including dependencies.
- Added `./gradlew clean shadowJar` command for JAR generation.
- Verified that `nickiminaj.jar` runs correctly using `java -jar`.

This update ensures that the chatbot can be easily executed as a
standalone application.

Commit:
Add showBarb() in UI class

Commit:
Add assertions to 'todo' case in Parser class

Commit:
Fix bugs in Parser by handling invalid inputs more elegantly

- Instead of using assert to terminate the programme, I used new private
methods so that nickiminaj can respond and tell you what to do with
an invalid input

Commit:
Rename all "Duke" instances to NickiMinaj

Commit:
Fix ArrayOutOfBounds error from parseTask() through new ErrorTask class

elizazaa committed:
Merge pull request #2 from elizazaa/branch-A-Assertions

Add Assertions to Parser for Robust Error Handling

Commit:
Add extension to view tasks on a particular date

Commit:
Resolved merge conflicts in Parser.java

Commit:
Merge branch 'master' of https://github.com/elizazaa/ip

* 'master' of https://github.com/elizazaa/ip:

Commit:
Add nickiminaj welcome message when entering

Commit:
Add better personality to nickiminaj's responses

Commit:
Improve nickiminaj's interface to be more like her personality

Commit:
Improve nickiminaj's personality

Commit:
Delete cardbi.png image

Commit:
Improve error handling

This was done by:
- Add assert statements
- nickiminaj's responses help the user type in the correct dates

Commit:
Add more javadoc

Commit:
Add more assert statements

Commit:
Remove extra line

Commit:
Add title to the app

Commit:
Add Ui.png to docs

Commit:
Add Readme

Commit:
Fix window's error of missing JavaFX dependencies
